topics/sw_concepts/sw_concept_slides.md

```diff
@@ -323,30 +323,25 @@ Design Patterns
 
 <https://github.com/iluwatar/java-design-patterns>
 
-Factory (method)
+Factory
 -------
 
-<https://github.com/iluwatar/java-design-patterns/blob/master/factory-method/README.md>
+* Alias: Factory Method, Simple Factory
+* Use Case:
+  * Logik für die Erstellung eines Objekts kapseln und vereinheitlichen.
+  * Wissen um konkrete Implementationen eines Interfaces zentralisieren.
```

Comment:
Bin nicht sicher, ob das so stimmt. Auch wenn man sich überlegt, was eine AbstractFactory macht. Aktuell sehen diese beiden Beispiele bei dir sehr ähnlich aus. Scheinbar ist der Unterschied auch nicht einfach zu erkennen/definieren: https://stackoverflow.com/questions/5739611/what-are-the-differences-between-abstract-factory-and-factory-design-patterns

Ein Teil ist vermutlich, dass es bei der "normalen" Factory um eine Methode einer Klasse handelt, die selber nicht die Factory ist. Nur die Methode ist eine "Factory". Beim AbstractFactory ist die Factory eine komplett unabhängige Klasse.

Reply:
Habe geschaut, dass im factory beispiel das Label verwendet wird, damit es ähnlich wie das AbstractFactory Beispiel aussieht.